### PR TITLE
Add prettier on precommit

### PR DESCRIPTION
**Pull request by miroslavstastny**
Prettier is run on precommit for staged files

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,4 @@ appsettings.json
 local.settings.json
 
 node_modules
+/.idea
```
Comment (layershifter) [inline .gitignore]: https://gist.github.com/subfuzion/db7f57fff2fb6998a16c
Use global `.gitignore` :)

File: package.json [hunk 2]
```diff
@@ -1,7 +1,35 @@
...
+    "lint-staged": "^8.1.5",
+    "prettier": "1.16.4"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.{js,css,json,md}": [
```
Comment (layershifter) [inline package.json]:
```suggestion
    "*.js": [
```
I suggest to leave only JS

File: package.json [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
     "mongodb": "3.1.13"
+  },
+  "devDependencies": {
+    "husky": "^1.3.1",
+    "lint-staged": "^8.1.5",
+    "prettier": "1.16.4"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
```
Comment (layershifter) [inline package.json:17]: https://github.com/okonet/lint-staged#frequently-asked-questions

```
"postcommit": "git update-index --again"
```

Comment (miroslavstastny) [reply package.json:17]: According to the FAQ not needed anymore

**Pull request merged**